src/fsharp/TastOps.fs

```diff
@@ -2707,12 +2718,36 @@ let fullDisplayTextOfValRefAsLayout   (vref:ValRef) =
         pathText ^^ SepL.dot ^^ wordL n
         //pathText +.+ vref.DisplayName
 
-
 let fullMangledPathToTyconRef (tcref:TyconRef) = 
     match tcref with 
     | ERefLocal _ -> (match tcref.PublicPath with None -> [| |] | Some pp -> pp.EnclosingPath)
     | ERefNonLocal nlr -> nlr.EnclosingMangledPath
-  
+    
+/// generates a name like 'System.IComparable<System.Int32>.Get'
+/// for types in the global namespace, `global is prepended (note the backtick)
+let qualifiedInterfaceImplementationName g (tt:TType) memberName =
```

Comment:
What happens for function types, nativeptrs, tuples, anon records (since it was added recently), and unions?

Reply:
Are they relevant? ``tt`` should be the interface that is being implemented, and tuples, anon records can't implement interfaces.

So I don't think these features intersect.

Reply:
`tt` won't be those types, you are right. I was referring to `TType_ToCodeLikeString` as those types will matter when you try to get type parameters. ex: `interface ITest<{| x: int |}>` , `interface ITest<nativeptr<int>>`.